substrate/frame/system/src/mock.rs

```diff
@@ -78,6 +78,14 @@ impl OnKilledAccount<u64> for RecordKilled {
 	}
 }
 
+#[derive(Debug, TypeInfo)]
+pub struct DefaultNonceProvider;
```

Comment:
Docs what it does please and why it is "Default".

Reply:
Oh yea 🙈  I did not realize that this was just adding the new type, not specifically for the nonce thing.